src/sentry/migrations/0166_create_notificationsetting_table.py

```diff
@@ -0,0 +1,166 @@
+                        choices=[
+                            (
+                                sentry.models.notificationsetting.NotificationSettingTypes(0),
+                                "default",
+                            ),
+                            (
+                                sentry.models.notificationsetting.NotificationSettingTypes(10),
+                                "deploy",
+                            ),
+                            (
+                                sentry.models.notificationsetting.NotificationSettingTypes(20),
+                                "issue",
+                            ),
+                            (
+                                sentry.models.notificationsetting.NotificationSettingTypes(30),
+                                "workflow",
+                            ),
+                        ]
+                    ),
+                ),
+                (
+                    "value",
+                    sentry.db.models.fields.bounded.BoundedPositiveIntegerField(
+                        choices=[
+                            (
+                                sentry.models.notificationsetting.NotificationSettingOptionValues(
+                                    0
+                                ),
+                                "default",
+                            ),
+                            (
+                                sentry.models.notificationsetting.NotificationSettingOptionValues(
+                                    10
+                                ),
+                                "off",
+                            ),
+                            (
+                                sentry.models.notificationsetting.NotificationSettingOptionValues(
+                                    20
+                                ),
+                                "on",
+                            ),
+                            (
+                                sentry.models.notificationsetting.NotificationSettingOptionValues(
+                                    30
+                                ),
+                                "subscribe_only",
+                            ),
+                            (
+                                sentry.models.notificationsetting.NotificationSettingOptionValues(
+                                    40
+                                ),
+                                "committed_only",
+                            ),
+                        ]
+                    ),
+                ),
+            ],
+            options={
+                "db_table": "sentry_notificationsetting",
+            },
+        ),
+        migrations.AlterUniqueTogether(
+            name="notificationsetting",
+            unique_together={
+                (
+                    "scope_type",
+                    "scope_identifier",
+                    "target_type",
+                    "target_identifier",
```

Comment:
Do you know what some of your more common queries will look like based on how useroption is used? That could help with adding additional indexes you might need.